### PR TITLE
Build an end-to-end pipeline example using custom splits

### PR DESCRIPTION
**Pull request by tinally**
# Build an end-to-end pipeline example using custom splits for ExampleGen and its downstream components.
The structure of the pipeline output is:
Csvexamplegen -> examples -> evaluating, testing, training
Evaluator -> blessing, evaluation
ExampleValidator -> anomalies -> training, evaluating
Pusher -> pushed_model
SchemaGen -> schema -> schema.pbtxt
StatisticsGen -> statistics -> evaluating, training
Trainer -> model, model_run
Transform -> transformed_examples -> evaluating, training
Tuner -> best_hyperparameter

### CODE REVIEW

File: tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                     metadata_path: Text, enable_tuning: bool,
+                     beam_pipeline_args: List[Text]) -> pipeline.Pipeline:
+  """Implements the Iris flowers pipeline with TFX."""
+  examples = external_input(data_root)
+
+  output = example_gen_pb2.Output(
+      split_config=example_gen_pb2.SplitConfig(splits=[
+          example_gen_pb2.SplitConfig.Split(name='training', hash_buckets=3),
+          example_gen_pb2.SplitConfig.Split(name='evaluating', hash_buckets=2),
+          example_gen_pb2.SplitConfig.Split(name='testing', hash_buckets=1)
```
Comment (1025KB) [inline tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py:94]: could you check the output file size to see if it basically align with 3:2:1

File: tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+# Transform and Trainer both require user-defined functions to run successfully.
+_module_file = os.path.join(_iris_root, 'iris_utils_native_keras.py')
+# Path which can be listened to by the model server.  Pusher will output the
+# trained model here.
+_serving_model_dir = os.path.join(_iris_root, 'serving_model', _pipeline_name)
+
+# Directory and data locations.  This example assumes all of the flowers
+# example code and metadata library is relative to $HOME, but you can store
+# these files anywhere on your local filesystem.
+_tfx_root = os.path.join(os.environ['HOME'], 'tfx')
```
Comment (1025KB) [inline tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py:68]: let's paste the output tfx/ structure

File: tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      exclude_splits=['testing'])
+
+  # Performs transformations and feature engineering in training and serving.
+  transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=schema_gen.outputs['schema'],
+      module_file=module_file,
+      splits_config=transform_pb2.SplitsConfig(analyze=['training'],
+                                               transform=['training',
+                                                          'evaluating']))
```
Comment (1025KB) [inline tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py:122]: FYI, transform is just submitted, you can download tfx head and install tfx head to test now

Comment (tinally) [reply tfx/examples/iris/iris_pipeline_native_keras_custom_splits.py:122]: yes, I was planning to wait until transform is merged and test it again

**Comment by 1025KB** [comment]
Thanks for e2e verification, looks good

**Pull request closed without merging**